RadixWallet/Features/AssetTransferFeature/Components/TransferAccountList/ReceivingAccount/Asset/FungibleResourceAsset+Reducer.swift

```diff
@@ -77,6 +110,61 @@ public struct FungibleResourceAsset: Sendable, FeatureReducer {
 
 		case .removeTapped:
 			return .send(.delegate(.removed))
+
+		case let .alertAction(action):
+			state.alert = nil
+			switch action {
+			case let .presented(.chooseXRDAmountAlert(.deductFee(amount))),
+			     let .presented(.chooseXRDAmountAlert(.sendAll(amount))),
+			     let .presented(.needsToPayFeeFromOtherAccount(.confirm(amount))):
+				state.transferAmount = amount
+				state.transferAmountStr = amount.formattedPlain(useGroupingSeparator: false)
+				return .send(.delegate(.amountChanged))
+
+			case .presented(.needsToPayFeeFromOtherAccount(.cancel)):
+				return .none
+
+			case .dismiss:
+				return .none
+			}
 		}
 	}
 }
+
+extension AlertState where Action == FungibleResourceAsset.ViewAction.AlertAction {
+	fileprivate static func chooseXRDAmount(feeDeductedAmount: RETDecimal, maxAmount: RETDecimal) -> Self {
+		.init(
+			title: .init(L10n.AssetTransfer.MaxAmountDialog.title),
+			message: .init(L10n.AssetTransfer.MaxAmountDialog.body),
+			buttons:
+			[
+				.default(
+					.init(L10n.AssetTransfer.MaxAmountDialog.saveXrdForFeeButton(feeDeductedAmount.formatted())),
```

Comment:
how does this look when you have 100 million XRD? (🐋)? The size of the button title will be tiny...?

Reply:
100 million XRD will be formatted as 100M though, thanks to @kugel3 formatter